source/lac/trilinos_solver.cc

```diff
@@ -918,6 +918,35 @@ namespace TrilinosWrappers
     do_solve();
   }
 
+
+  void SolverDirect::solve (dealii::LinearAlgebra::distributed::Vector<double>       &x,
+                            const dealii::LinearAlgebra::distributed::Vector<double> &b)
+  {
+    Epetra_Vector ep_x (View, linear_problem->GetOperator()->OperatorDomainMap(), x.begin());
+    Epetra_Vector ep_b (View, linear_problem->GetOperator()->OperatorRangeMap(), const_cast<double *>(b.begin()));
+
+    // Assign the empty LHS vector to the Epetra_LinearProblem object
+    linear_problem->SetLHS(&ep_x);
+
+    // Assign the RHS vector to the Epetra_LinearProblem object
+    linear_problem->SetRHS(&ep_b);
+
+    // Fetch return value of Amesos Solver functions
+    int ierr;
```

Comment:
Since you don't assign anything here, the comment is a bit funny. Maybe move the declaration of the variable down to its first use?